examples/opentelemetry-jaeger/src/main.rs

```diff
@@ -0,0 +1,77 @@
+//! Run jaeger
+//!
+//! ```not_rust
+//! docker run -d -p6831:6831/udp -p6832:6832/udp -p16686:16686 -p14268:14268 jaegertracing/all-in-one:latest
+//! ```
+//!
+//! Run the server
+//!
+//! ```not_rust
+//! cargo run -p example-opentelemetry-jaeger
+//! ```
+
+use axum::{http::Request, routing::get, Router};
+use axum_extra::middleware::opentelemtry_tracing_layer;
+use std::net::SocketAddr;
+use tower::ServiceBuilder;
+use tower_http::request_id::{RequestId, SetRequestIdLayer};
+use tracing_subscriber::prelude::*;
+
+#[tokio::main]
+async fn main() {
+    // set the RUST_LOG, if it hasn't been explicitly defined
+    if std::env::var_os("RUST_LOG").is_none() {
+        std::env::set_var(
+            "RUST_LOG",
+            "axum_extra=debug,example_opentelemetry_jaeger=debug,tower_http=debug",
+        )
+    }
```

Comment:
Since the future of `set_var` is unclear (https://github.com/rust-lang/rust/issues/90308), I would recommend instead replacing the `try_from_default_env().unwrap()` below with a `try_from_default_env().unwrap_or_else()` where you create the `EnvFilter` from this string.

Reply:
I'll cross that bridge when we get there. I don't think there is any problem with how we're using it now.

Reply:
I really recommend the approach that jplatte suggested: that is the approach tracing pushes you to adopt.